src/applications/simple-forms/mock-simple-forms-patterns/arrayBuilder/components/ArrayBuilderChapter.jsx

```diff
@@ -0,0 +1,373 @@
+ * @property {function(FormConfigPage): FormConfigPage} [itemLastPage] Last page for the item that will navigate back to the summary page
+ * @property {function(FormConfigPage): FormConfigPage} [itemSinglePage] Use this if there's only one page
+ */
+
+/**
+ * @typedef {Object} ArrayBuilderOptions
+ * @property {string} noun e.g. `employer`
+ * @property {string} arrayPath e.g. `employers`
+ * @property {string} nextChapterPath e.g. `/next-chapter`
+ * @property {string} [reviewPath] e.g. `/review-and-submit`
+ * @property {(item) => boolean} [isIncompleteItem] Will display error on the cards if item is incomplete. e.g. `item => !item?.name`
+ */
+
+/**
+ * @typedef {FormConfigChapter & {
+ *   options: ArrayBuilderOptions
+ * }} ArrayBuilderConfig
+ */
+
+function getBasePath(path) {
+  const basePath = path.split('/:')[0];
+  return basePath.startsWith('/') ? basePath : `/${basePath}`;
+}
+
+function throwErrorPage(pageType, option) {
+  throw new Error(
+    `arrayBuilderChapter \`pageBuilder.${pageType}()\` must include \`${option}\` property like this: ` +
+      `\`arrayBuilderChapter(pageBuilder => ({ examplePage: pageBuilder.${pageType}({ ${option}: ... }) }))\``,
+  );
+}
+
+function throwErrorChapter(option) {
+  throw new Error(
+    `arrayBuilderChapter must include \`options.${option}\` property`,
+  );
+}
+
+function throwErrorNoConfigFunction() {
+  throw new Error(
+    'arrayBuilderChapter must include a config function like this `arrayBuilderChapter(pageBuilder => { ... })`',
+  );
+}
+
+function throwErrorNoOptions() {
+  throw new Error(
+    'arrayBuilderChapter must include an `options` property like this: ' +
+      '`arrayBuilderChapter(pageBuilder => ({ options: { ... } }))`',
+  );
+}
+
+function throwMissingYesNoField() {
+  throw new Error(
+    "arrayBuilderChapter `pageBuilder.summaryPage()` must include a `uiSchema` that has a property with a `'ui:webComponentField': YesNoField`",
+  );
+}
+
+function throwIncorrectItemPath() {
+  throw new Error(
+    'arrayBuilderChapter item pages must include a `path` property that ends with `/:index`',
+  );
+}
+
+function verifyRequiredOptions(pageType, requiredOpts, objectToCheck) {
+  requiredOpts.forEach(option => {
+    if (objectToCheck[option] === undefined) {
+      if (pageType === 'chapter') {
+        throwErrorChapter(option);
+      } else {
+        throwErrorPage(pageType, option);
+      }
+    }
+  });
+}
+
+function determineYesNoField(uiSchema) {
+  let yesNoKey;
+  for (const key of Object.keys(uiSchema)) {
+    if (uiSchema[key]['ui:webComponentField'] === YesNoField) {
+      yesNoKey = key;
+    }
+  }
+  return yesNoKey;
+}
+
+/**
+ * arrayBuilderChapter
+ *
+ * Example:
+ * ```
+ * chapterName: arrayBuilderChapter(pageBuilder => {
+ *   title: 'Employment history',
+ *   options: {
+ *     arrayPath: 'employers',
+ *     noun: 'employer',
+ *     nextChapterPath: '/next-chapter',
```

Comment:
So, the array pages _always have_ to be in their own chapter? What happens if the chapter isn't visible because all the pages are hidden?

And we already discussed the path's needing an `:index` included...

Reply:
There's some platform code that determines where the "Continue" button will navigate to, this work is building off of a custom component, so I think you can use the `goForward` callback.